### ISSUE TITLE
Possible encoding issue

### ISSUE THREAD

**Issue by precipice** [justinweiss/dragongoclient#21]
I made a comment during a move on an in-progress game. The comment started:

```No—I ...``` (and then there was more of the comment)

My opponent replied (as seen on DGS' site):

```the beginning of your last post was: Noâ€”I```

In the DGS app, his comment reads:

```the beginning of your last post was: No—I``` (and no more of the comment came through for him)

Generally looks like a comment encoding issue to me; I'll take a look at this in a bit.

**Comment by precipice** [comment]
Oh, interestingly, if I look at the comment history on DGS's site, the full comment made it through, just with a weird formatting of the em-dash, which apparently causes the DGS app to truncate the comment.

**Comment by justinweiss** [comment]
Yep, that's usually a double-encoding issue - You send UTF-8, it's parsed as Windows-1252, and then re-encoded to UTF-8. 

Funny enough, I wrote about this category of problems a little while ago: https://www.justinweiss.com/articles/3-steps-to-fix-encoding-problems-in-ruby/

**Comment by precipice** [comment]
I'm seeing this again with the current build, on iOS 11.1, iPhone X, Dragon Go Client 2.1. The issue appears to be "smart" apostrophes. They're rendered on dragongoserver.net as `â€™` (e.g., in `Itâ€™s`). I'll take a look to see if I can find the problem.

**Comment by justinweiss** [comment]
Hm, interesting. Encoding has been a huge issue because of all of the different languages DGS supports, and I haven't been able to find a way to handle them that's totally consistent.
